x-pack/test/cloud_security_posture_functional/pages/custom_role_access.ts

```diff
@@ -0,0 +1,310 @@
+    });
+
+    describe('Access with custom roles', async () => {
+      before(async () => {
+        await security.role.create(VALID_CUSTOM_ROLE.name, {
+          elasticsearch: VALID_CUSTOM_ROLE.elasticsearch,
+          kibana: VALID_CUSTOM_ROLE.kibana,
+        });
+        await security.user.create(VALID_USER.name, {
+          password: VALID_USER.password,
+          roles: [VALID_CUSTOM_ROLE.name],
+        });
+        await navigateToHomePage;
+        await pageObjects.security.forceLogout();
+        await pageObjects.security.login(VALID_USER.name, VALID_USER.password, {
+          expectSpaceSelector: false,
+        });
+      });
+
+      it('Access to the dashboard page', async () => {
+        await cspDashboard.navigateToComplianceDashboardPage();
+        await retry.waitFor(
+          'Cloud posture integration dashboard to be displayed',
+          async () => !!dashboard.getIntegrationDashboardContainer()
+        );
+        const scoreElement = await dashboard.getKubernetesComplianceScore();
+
+        expect((await scoreElement.getVisibleText()) === '50%').to.be(true); // based on the ingeated findings
+      });
+
+      it('Access to the findings page', async () => {
+        await findings.navigateToLatestFindingsPage();
+        pageObjects.header.waitUntilLoadingHasFinished();
+
+        expect(await latestFindingsTable.getRowsCount()).to.be.greaterThan(0);
+      });
+
+      it('Access to the benchmark page', async () => {
+        await benchmark.navigateToBenchnmarkPage();
+
+        expect(await benchmark.benchmarkPage.doesBenchmarkTableExists());
+      });
+
+      it('Access to the rules page', async () => {
+        await rule.navigateToRulePage('cis_k8s', '1.0.1');
+
+        expect(await rule.rulePage.toggleBulkActionButton());
+      });
+    });
+
+    describe('Custom Roles Access - Valid Role', async () => {
+      before(async () => {
+        await security.role.create(INVALID_CUSTOM_ROLE.name, {
+          elasticsearch: INVALID_CUSTOM_ROLE.elasticsearch,
+          kibana: INVALID_CUSTOM_ROLE.kibana,
+        });
+        await security.user.create(INVALID_USER.name, {
+          password: INVALID_USER.password,
+          roles: [INVALID_CUSTOM_ROLE.name],
+        });
+
+        await navigateToHomePage;
+        await pageObjects.security.forceLogout();
+        await pageObjects.security.login(INVALID_USER.name, INVALID_USER.password, {
+          expectSpaceSelector: false,
+        });
+      });
+
+      it('Access findings page', async () => {
+        await findings.navigateToLatestFindingsPage();
+
+        pageObjects.header.waitUntilLoadingHasFinished();
+        expect(await findings.getUnprivilegedPrompt());
+      });
+
+      // Blocked by https://github.com/elastic/kibana/issues/184621
+      // it('Fail to access the dashboard page', async () => {
+
+      // });
+
+      // it('Access to the benchmark page', async () => {
+
+      // });
```

Comment:
please add `todo` keyword and run the script to generate the needed files after those changes

1. todo example:
```
    //   it('todo - displays accurate summary compliance score', async () => {});
```

2. the script:
```
node x-pack/plugins/cloud_security_posture/common/scripts/get_tests.js
```

Reply:
done.